libraries/GCS_MAVLink/MAVLink_routing.cpp

```diff
@@ -140,6 +140,13 @@ bool MAVLink_routing::check_and_forward(GCS_MAVLINK &in_link, const mavlink_mess
     int16_t target_component = -1;
     get_targets(msg, target_system, target_component);
 
+    // handle messages from a mavlink receiver as if it had target_system = our system, target_component = 0
+    if ((msg.compid == MAV_COMP_ID_TELEMETRY_RADIO) &&
+        (msg.msgid == MAVLINK_MSG_ID_RADIO_RC_CHANNELS)) {
+        target_system = mavlink_system.sysid;
+        target_component = 0;
+    }
```

Comment:
This doesn't look great

Add fields to the packet for target system/component.

Reply:
concerning targets, pl see post below.